src/gui/sdlmain.cpp

```diff
@@ -2782,68 +2779,60 @@ static bool wants_stretched_pixels()
 	return render_section->Get_bool("aspect");
 }
 
-static SDL_Point remove_stretched_aspect(const SDL_Point &size)
+SDL_Point calc_nearest_aspect_corrected_dimensions(const SDL_Point& source_dimensions,
+                                                   const SDL_Point& aspect_ratios,
+                                                   const int in_multiples_of)
 {
-	return {size.x, ceil_sdivide(size.y * 5, 6)};
+	auto to_nearest_multiple = [=](const int source_dimension,
+	                               const int aspect_ratio) {
+		assert(aspect_ratio > 0);
+		assert(in_multiples_of > 0);
+		const auto aspect_multiple = aspect_ratio * in_multiples_of;
+		assert(source_dimension > aspect_multiple);
+		return (source_dimension / aspect_multiple) * aspect_multiple;
+	};
+
+	auto width = to_nearest_multiple(source_dimensions.x,
+	                                 aspect_ratios.x);
+
+	auto height = to_nearest_multiple(source_dimensions.y,
+	                                  aspect_ratios.y);
+
+	const auto width_with_ratio  = width * aspect_ratios.y;
+	const auto height_with_ratio = height * aspect_ratios.x;
+
+	// screen is landscape, so bound the width using the height
+	if (width_with_ratio > height_with_ratio) {
+		width = height_with_ratio / aspect_ratios.y;
+	}
+	// screen is portrait, so bound the height using the width
+	else if (height_with_ratio > width_with_ratio) {
+		height = width_with_ratio / aspect_ratios.x;
+	}
+
+	return {width, height};
```

Comment:
Same comments as above. Can't we extract the common logic to some common files instead of copy-pasting it?